week4/week4.playground/Contents.swift

```diff
@@ -0,0 +1,203 @@
+import Cocoa
+
+let zeroToTwenty = Array(0...20)
+func average(nums: [Int]?) {
+  if let nonOptionalNums = nums {
+    if nonOptionalNums.count > 0 {
+      let total = nonOptionalNums.reduce(0, +)
+      let average = total/nonOptionalNums.count
+      print("The average of the values in the array is \(average)")
+    } else {
+      print("The array is nil. Calculating the average is impossible.")
+    }
+  }
+}
+
+average(nums: zeroToTwenty)
+let optionalIntArray: [Int]? = []
+average(nums: optionalIntArray)
+
+
+// a. create overload functions using average method and an array
+
+func average(numbers: [Int]?) -> Int {
+  var average: Int = 0
+  if let numbers = numbers, numbers.count > 0 {
+    average = numbers.reduce(0, +)/numbers.count
+  }
+  return average
+}
+
+print("The average of new overloaded function passing a default value for the parameter is \(average(numbers: zeroToTwenty))")
+
+func average() -> Int {
+  let average = zeroToTwenty.reduce(0, +)/zeroToTwenty.count
+  return average
+}
+let avg = average()
+print("the average of first 20 numbers are \(avg)")
+
+//  Create an enum called Animal that has at least five animals. Next, make a function called theSoundMadeBy that has a parameter of type Animal. This function should output the sound that the animal makes. For example, if the Animal passed is a cow, the function should output, “A cow goes moooo.”
+// Hint: Do not use if statements to complete this section.
+// Call the function twice, sending a different Animal each time.
+
+enum Animal: String {
```

Comment:
You are conforming your enum to a `String`, but you're not using that anywhere so this is unnecessary.